metricbeat/module/elasticsearch/ccr/ccr.go

```diff
@@ -56,15 +56,19 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 
 // Fetch gathers stats for each follower shard from the _ccr/stats API
 func (m *MetricSet) Fetch(r mb.ReporterV2) error {
-	isMaster, err := elasticsearch.IsMaster(m.HTTP, m.GetServiceURI())
-	if err != nil {
-		return errors.Wrap(err, "error determining if connected Elasticsearch node is master")
-	}
+	// If we're talking to a set of ES nodes directly, only collect stats from the master node so
+	// we don't collect the same stats from every node and end up duplicating them.
+	if m.HostsMode == elasticsearch.HostsModeNode {
+		isMaster, err := elasticsearch.IsMaster(m.HTTP, m.GetServiceURI())
+		if err != nil {
+			return errors.Wrap(err, "error determining if connected Elasticsearch node is master")
+		}
 
-	// Not master, no event sent
-	if !isMaster {
-		m.Logger().Debug("trying to fetch ccr stats from a non-master node")
-		return nil
+		// Not master, no event sent
+		if !isMaster {
+			m.Logger().Debug("trying to fetch ccr stats from a non-master node")
+			return nil
+		}
 	}
```

Comment:
this logic repeats across the board, I wonder if it should go into a helper function

Reply:
Refactored in 348c9b9.